docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
+   `PAYMENT_METHOD`, `CATEGORY` input.
+3. The `AddCommand.addExpenditure()` object instantiates a new `Expenditure` object with the aforementioned 5 fields and adds them
+   into the `ExpenditureList`.
+4. The `AddCommand.addExpenditure()` object prints a list to show the user what it has saved.
+5. The `AddCommand.addExpenditure()` returns to `AddCommand`.
+
+#### AddCommand Design Considerations
+Aspect: How to ask user for the fields of input.
+* Alternative 1 (current choice): User is asked to put in all fields at once, separated using flags.
+    * Pros: Faster input, user can enter an expenditure using a single input.
+    * Cons: User must be able to remember all the flags and its sequence.
+
+* Alternative 2: User is asked iteratively to put in all fields, prompted by a message after each input.
+    * Pros: Beginner friendly, easily understandable, no need to remember flags.
+    * Cons: Slower, implementation when user is familiar with the application.
+
+### CalculateCommand feature
+
+To enable users to view their finances in a more meaningfully, MindMyMoney does calculations to present financial data
+that is actionable for the users.
+
+![calculate_command_sequence_diagram](images/gif_loading.gif)
+
+### ListCommand feature
+The source code can be found in [`ListCommand.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command/ListCommand.java)  
+
+MindMyMoney allow users to view their current list of added expenditures. 
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when listing.  
+
+![list_command_sequence_diagram](images/List_Command_Sequence_Diagram.png)
+Fig 13 - List Command Sequence Diagram
+
+#### ListCommand design considerations
+Aspect: How to make the `ListCommand` easily tested using JUnit testing.
+* Alternative 1 (current choice): Abstract the conversion of `Expenditure` to `String` in a separate` ListCommand.listToString()` method.
+  * Pros: Easily tested using JUnit testing, by checking the String that the `ListCommand.listToString()` method returns.
+  * Cons: Added layer of abstraction that may be deemed redundant.
+
+* Alternative 2: Print directly in the `ListCommand.executeCommand()` method .
+  * Pros: Easily implemented with lesser lines of code as the code is minimalist.
+  * Cons: JUnit testing would require I/O redirection prior to checking the output matches expectations.
+
+
+
+## Appendix Requirements
+  
+  
+### Product scope
+
+**Target user profile**
+- a student beginner who wants to start managing their finances
+- possess an income
+- prefer desktop application
+- is proficient in CLI
+- can type fast
+- prefers typing to mouse interactions
+
+**Value proposition**
+Manage finances containing multiple payment methods faster than a typical mouse/GUI driven app.
+
+### User Stories
```

Comment:
Would it be possible to add more user stories that could cover what kind of users the features are intended for?